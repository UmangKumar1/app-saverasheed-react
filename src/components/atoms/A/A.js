import PropTypes from 'prop-types'
import styled from 'react-emotion'

const { string, any } = PropTypes

const A = styled('a')({
  textDecoration: 'none',
  color: 'inherit',
})

A.propTypes = {
  children: any,
  href: string.isRequired,
}

export default A
