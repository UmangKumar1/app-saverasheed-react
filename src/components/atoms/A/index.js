export { default, default as A } from './A'
