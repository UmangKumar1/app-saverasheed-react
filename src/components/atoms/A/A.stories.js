import React from 'react'
import { storiesOf } from '@storybook/react'
import A from '.'

storiesOf('A', module).add('with text', () => (
  <A href="https://google.com">Save Rasheed</A>
))
