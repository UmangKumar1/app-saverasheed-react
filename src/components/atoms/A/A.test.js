import React from 'react'
import { shallow } from 'enzyme'
import A from '.'

const wrap = (props = {}) => shallow(<A {...props} />)

describe('A', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({ children: 'A', href: 'https://google.com' })
    expect(wrapper).toMatchSnapshot()
  })
})
