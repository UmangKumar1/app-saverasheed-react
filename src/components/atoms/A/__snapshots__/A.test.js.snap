// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(a)
    href="https://google.com"
  >
    A
  </Styled(a)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "A",
      "className": "css-nn640c",
      "href": "https://google.com",
    },
    "ref": null,
    "rendered": "A",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "A",
        "className": "css-nn640c",
        "href": "https://google.com",
      },
      "ref": null,
      "rendered": "A",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
